testing/jormungandr-integration-tests/src/networking/p2p.rs

```diff
@@ -278,7 +302,10 @@ pub fn node_trust_itself() {
 
     assert!(network_controller
         .expect_spawn_failed(
-            params(CLIENT).trusted_peers(vec![peer]),
+            network_controller
+                .spawn_params(CLIENT)
+                .unwrap()
+                .trusted_peers(vec![peer]),
             "unable to reach peer for initial bootstrap"
         )
```

Comment:
These nested calls to controller methods make me suspect they could be redesigned as a proper chain, like:

```rust
network_controller
    .spawn_params(CLIENT)
    .unwrap()
    .trusted_peers()
    .expect_failed("unable to reach peer for initial bootstrap")
```

Reply:
This requires non trivial amount of work, might be worth investigating but I don't think it fits this PR